cli/internal/textutils_test.go

```diff
@@ -27,3 +28,49 @@ func Test_JSONPathToJSONObj(t *testing.T) {
 		})
 	}
 }
+
+func TestUnfoldMap(t *testing.T) {
+	type args struct {
+		toAdd map[string]string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    map[string]interface{}
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name: "minimal event data",
+			args: args{
+				toAdd: map[string]string{"project": "pr", "stage": "st", "service": "sv"},
+			},
+			want:    map[string]interface{}{"project": "pr", "stage": "st", "service": "sv"},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "valid additional fields",
+			args: args{
+				toAdd: map[string]string{"project": "pr", "stage": "st", "service": "sv", "a.b": "value", "b.a": "othervalue"},
```

Comment:
maybe also this to be extra sure: 
```suggestion
				toAdd: map[string]string{"project": "pr", "stage": "st", "service": "sv", "a.b": "value", "b.a": "othervalue", "b.a.c": "myotherCvalue"},
```

Reply:
as "b.a" is already bound to a value (string "othervalue") this case is not possible and marks the limit of the solution implemented here. Nonetheless I can more specifically test this "failure" case.